src/main/java/hudson/plugins/pmd/PmdPublisher.java

```diff
@@ -25,14 +24,20 @@
  * @author Ulli Hafner
  */
 public class PmdPublisher extends HealthAwarePublisher {
+
     private static final long serialVersionUID = 6711252664481150129L;
```

Comment:
Capture a sample instance in a `byte[]` from before the change and then use that as test data, verifying that the `readResolve()` does the correct transformation / populating of fields... but you'd need a test like that whenever you add a `readResolve()` anyway